float64[16] right_trans 
float64[16] left_trans 
---
float64[8] bump_vals